dwave/optimization/src/nodes/mathematical.cpp

```diff
@@ -1776,6 +1784,10 @@ double UnaryOpNode<functional::square<double>>::min() const {
     }
     return min * min;
 }
+template <>
+double UnaryOpNode<functional::square_root<double>>::min() const {
+    return std::sqrt(array_ptr_->min());
```

Comment:
As I mentioned elsewhere, the only thing to decide is how to handle negative numbers... Are we even going to accept the predecessor to have
```cpp
array_ptr_->min() < 0
```
?

Reply:
Up to you guys.  Like I mentioned in the other thread, I'm planning on letting std::sqrt determine the behviour.  If you want something different, just let me know.

Reply:
We want to disallow it. We're most interested in following NumPy while not allowing `nan`. So we need to require `array_ptr->min() >= 0`.

Reply:
What's your behaviour expectation?

Reply:
We should check `predecessor->min() >= 0` on construction and raise an ``std::invalid_argument` error if no. That means the the predecessor will always output a non-negative number and the default `std::sqrt` behavior will be correct.

Reply:
```c++
template <class UnaryOp>
UnaryOpNode<UnaryOp>::UnaryOpNode(ArrayNode* node_ptr)
        : ArrayOutputMixin(node_ptr->shape()), array_ptr_(node_ptr) {
    if constexpr (std::is_same<UnaryOp, functional::square_root<double>>::value) {
        // raise error message here.
    }
    add_predecessor(node_ptr);
}
```